h2o-core/src/main/java/hex/Model.java

```diff
@@ -820,6 +832,45 @@ public void printTwoDimTables(StringBuilder sb, Object o) {
   } // Output
 
   protected String[][] scoringDomains() { return _output._domains; }
+  
+  public void addTargetEncodingMap(Map<String, Frame> encodingMap) {
+    IcedHashMap<String, Map<String, TEComponents>> transformedEncodingMap = new IcedHashMap<>();
+    for (Map.Entry<String, Frame> entry : encodingMap.entrySet()) {
+      String key = entry.getKey();
+      Frame encodingsForParticularColumn = entry.getValue();
+      IcedHashMap<String, TEComponents> table = new FrameToTETable().doAll(encodingsForParticularColumn).getResult().table;
```

Comment:
can't all columns be done in parallel by using `dfork` instead?

Reply:
@seb-h2o I was thinking about parallelisation on a encoding map level. But isn't doAll will span over multiple threads and complete each individual encoding map in a parallel fashion and utilising resources/threads efficiently? Because parallelising by TE columns as well might cause unnecessary overhead on splitting into more forks. There is no right answer to this unless we know how many cores we have. This is also not IO operation as we map/reduce in memory so rule of thumb for that case is that we don't wan't to split our task into number of subtasks that is greater than number of cores plus some adjustment. @michalkurka wdyt?

Reply:
afaik, it won't create a new thread for each fork, this is implemented by lightweight threads, ie. queues on top of predefined amount of threads: therefore it should always be faster than sequential loop.

Reply:
@seb-h2o what about `doAll` mapReduce? it is not using those lightweight threads? If it is also using them and as you mentioned it operates on top of predefined amount of threads than we still have to be reasonable in splitting tasks as there should be overhead on managing those queues.

Reply:
@deil87 `doAll().getResult()` is blocking, so your `for` loop is completely sequential.

Reply:
and `overhead on managing those queues` should be really really minimal

Reply:
@seb-h2o I know that. But doAll itself is doing parallelisation. If it is enough parallelisation on mapReduce level of `doAll` based on chunks layout then maybe there is no need to parallelise on for loop level. Why don't we run automl models all in parallel then? because it would only hurt performance because of the overhead of swapping between tasks in lightweight queue or in thread pool or whatever.